Feature/virtualizer new approach

GiulioRomualdi opened the pull request:
This PR is still in draft and before merging it we should clean a bit the code and improve the performances on iCub

This PR is only to keep track of the activity.  This code runs in icub-virtualizer laptop

GiulioRomualdi commented:
I cleaned the history and I also removed hardcoded 5 for the walking linear velocity.

The PR is ready to be reviewed @S-Dafarra and @kouroshD

S-Dafarra reviewed (approved):
LGTM. Just to understand, the connection of the goal port to the walking controller needs to be done manually, right?

GiulioRomualdi commented:
Yes as before. In theory, this should be done by the application.

kouroshD commented on app/robots/iCubGenova09/virtualizerConfig.ini:
```diff
@@ -2,23 +2,23 @@ name            virtualizer
 robotOrientationPort_name     /robotOrientation:i
-rpcWalkingPort_name           /walkingRpc
-scale_X                       3.0
+rpcWalkingPort_name           /goal:o
```
Please update the `rpcWalkingPort_name` with some name like `robotGoalPort_name` or sth similar.

kouroshD replied:
Please update the port name in all the xml files.

kouroshD replied:
Also, please also put the same port name for all the robot in the `.ini` files.
Thanks :)

kouroshD reviewed (changes requested):
Thank you @GiulioRomualdi  for the PR. I have added some minor comments.

kouroshD commented on app/robots/iCubGenova09/virtualizerConfig.ini:
```diff
@@ -2,23 +2,23 @@ name            virtualizer
 robotOrientationPort_name     /robotOrientation:i
-rpcWalkingPort_name           /walkingRpc
-scale_X                       3.0
+rpcWalkingPort_name           /goal:o
```
```suggestion
goalWalkingPort_name           /goal:o
```

kouroshD commented on modules/Virtualizer_module/src/VirtualizerModule.cpp:
```diff
@@ -261,12 +261,18 @@ bool VirtualizerModule::configure(yarp::os::ResourceFinder& rf)
+        return false;
+    }
+
     if (!YarpHelper::getStringFromSearchable(rf, "rpcWalkingPort_name", portName))
```
```suggestion
    if (!YarpHelper::getStringFromSearchable(rf, "goalWalkingPort_name", portName))
```

kouroshD commented:
@GiulioRomualdi  let me know if there are some updates on this?

kouroshD commented:
@GiulioRomualdi  As I wrote you before, I have updated the configuration files; I rebased it also on top of devel :) .
Let me know if it is OK for you, so we can merge it.

GiulioRomualdi commented:
> @GiulioRomualdi As I wrote you before, I have updated the configuration files; I rebased it also on top of devel :) .
> Please have a look as well, and let me know if it is OK for you, so we can merge it.

It's ok! You can merge it. Thank you @kouroshD
